auslib/db.py

```diff
@@ -662,7 +672,51 @@ def getRecentChanges(self, limit=10, transaction=None):
         return self.history.select(transaction=transaction, limit=limit, order_by=self.history.timestamp.desc())
 
 
-class History(AUSTable):
+class GCSHistory:
+    def __init__(self, db, dialect, metadata, baseTable, buckets, identifier_column, data_column):
+        self.buckets = buckets
+        self.identifier_column = identifier_column
+        self.data_column = data_column
+
+    def _getBucket(self, identifier):
+        for substring, bucket in self.buckets.items():
+            if substring in identifier:
+                return bucket
+        else:
+            raise KeyError("Couldn't find bucket to place {} history in.".format(identifier))
+
+    def forInsert(self, insertedKeys, columns, changed_by, trans):
+        timestamp = getMillisecondTimestamp()
+        identifier = columns.get(self.identifier_column)
+        for data_version, ts, data in ((None, timestamp - 1, ""), (columns.get("data_version"), timestamp, json.dumps(columns[self.data_column]))):
+            bname = "{}/{}-{}-{}.json".format(identifier, data_version, ts, changed_by)
```

Comment:
It's always 1 right now AFAICT (because of https://github.com/mozilla/balrog/blob/a7437b21bd3576c2ec8f8680264d57b24f0cd8e1/auslib/db.py#L408), but since `AUSTable` is responsible for deciding it, I thought it best to grab it from the args rather than make that assumption.

Reply:
Ok, sounds good.